core/org.wso2.carbon.user.core/src/main/java/org/wso2/carbon/user/core/util/DatabaseUtil.java

```diff
@@ -574,7 +606,11 @@ public static void closeConnection(Connection dbConnection) {
 
         if (dbConnection != null) {
             try {
-                dbConnection.close();
+				if (dbConnection != null && !dbConnection.isClosed()) {
```

Comment:
On line 609, "dbConnection != null" is redundant from line 608 and should be removed.

We should actually do 2 checks here.  Refer to [DBCP 1.4 API docs](http://commons.apache.org/proper/commons-dbcp/api-1.4/org/apache/commons/dbcp/BasicDataSource.html): 

1. if (datasource.isClosed()) // If true, this data source is closed and no more connections can be retrieved from this datasource.  We need to check this to see if its even possible to gain any more connections from the JDBC DataSource object.  So, call createRealmDataSource() above again with "datasource = createRealmDataSource(realmConfig);" and retry this whole method again (refactor to an overloaded exception handler method)
2. "else if (!dbConnection.isClosed())" // as is in the PR call dbConnection.close()"